### PR TITLE
feat: Remove editable checkbox from PrimaryColumnsControlV2 and update imports

### LINKED ISSUE #33888: Remove editable property in the table columns
Users get confused with show/hide and editable column properties.

[User testing link](https://app.usertesting.com/v/8ad33eab-deb5-405a-be84-e2b40ba0ecf2?encrypted_video_handle=9e4eec4f-914e-42a1-827f-45cbfd45219d&note_id=annotation-14967874&start=1198&shared_via=link&share_id=NIPsj9PbUR)

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
One of the findings in usability tests is that editable fields in table tend to confuse our users. With this PR, we attempt to remove editable option in table cols.

<ins>Summary of code changes</ins>
This PR consists of code changes to `PrimaryColumnsControlV2.tsx` to hide the editable checkboxes in the table. 
All cypress changes just make sure that the we use the colSettings view to make the column editable(earlier we used checkbox there.
* Removed unnecessary cypress command as well.

Fixes #33888
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Widget, @tag.Table, @tag.Binding"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9904610336>
> Commit: 890d9355b375e66e5e56a52464d8c0f19c759b9f
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9904610336&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Widget, @tag.Table, @tag.Binding`
> Spec:
> <hr>Fri, 12 Jul 2024 09:16:41 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **Refactor**
  - Improved inline editing functionality test by updating the method used to toggle column editability for better accuracy and maintainability.

### CODE REVIEW

**Comment by rahulbarwal** [comment]
@coderabbitai review

File: app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx [hunk 1]
```diff
@@ -234,7 +211,7 @@ class PrimaryColumnsControlV2 extends BaseControl<ControlProps, State> {
               renderComponent={(props: any) =>
                 DraggableListCard({
                   ...props,
-                  showCheckbox: true,
+                  showCheckbox: false,
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx]: Done

**Comment by ApekshaBhosale** [comment]
@coderabbitai review

**Comment by sagar-qa007** [comment]
@coderabbitai full review

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_1_spec.js [hunk 2]
```diff
@@ -17,174 +16,7 @@ describe(
...
-        cy.wait("@updateLayout");
-        cy.get(propPaneBack).click();
-        cy.get(`[data-rbd-draggable-id="step"] .t--card-checkbox input`).should(
-          data.expected,
-        );
-      });
-    });
-
-    it("8. should check that editable property is only available for Plain text & number columns", () => {
+    it("1. should check that editable property is only available for Plain text & number columns", () => {
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_1_spec.js:19]: Here I can see 7 test cases are removed. Any specific reason behind it?

Comment (rahulbarwal) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_1_spec.js:19]: Yes, so basically we are simply removing the behavior that these 7 tests were testing. 

We are removing checkboxes in the table column properties that make cols editable.

**Review by sagar-qa007 — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_3_spec.js [hunk 3]
```diff
@@ -1,13 +1,6 @@
...
 const widgetsPage = require("../../../../../locators/Widgets.json");
-import {
-  agHelper,
-  entityExplorer,
-  propPane,
-  table,
-  draggableWidgets,
-} from "../../../../../support/Objects/ObjectsCore";
-import { PROPERTY_SELECTOR } from "../../../../../locators/WidgetLocators";
+import { agHelper, table } from "../../../../../support/Objects/ObjectsCore";
```
[the remark this answers is not in the document]
Comment (jacquesikot) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_3_spec.js:3]: @rahulbarwal This makes sense...

File: app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx [hunk 4]
```diff
@@ -1,50 +1,39 @@
...
-import type { Placement } from "popper.js";
 import * as Sentry from "@sentry/react";
+import type { CodeEditorExpected } from "components/editorComponents/CodeEditor";
+import { EditorTheme } from "components/editorComponents/CodeEditor/EditorConfig";
+import EvaluatedValuePopup from "components/editorComponents/CodeEditor/EvaluatedValuePopup";
+import { DraggableListCard } from "components/propertyControls/DraggableListCard";
+import type { Indices } from "constants/Layers";
+import { Button } from "design-system";
+import type { DataTree } from "entities/DataTree/dataTreeTypes";
 import _, { toString } from "lodash";
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx]: Done.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_3_spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Inline_editing_3_spec.js]: Done.

**Review by jacquesikot — APPROVED** [review]

**Pull request merged**